typescript/core/src/parse.ts

```diff
@@ -417,3 +417,27 @@ function parseDataEnd(reader: Reader, recordLength: number): TypedMcapRecord {
     dataSectionCrc,
   };
 }
+
+export function monoParseMessage(reader: Reader): TypedMcapRecord | undefined | null {
```

Comment:
I think returning `TypedMcapRecord | undefined | null` is really odd. I would prefer to return `undefined` in all cases and let the slow path (parseRecord) handle length checks and error throwing.

Reply:
You're correct, differentiating between `undefined` & `null` isn't yet required or leveraged, maybe even not necessary.

I was thinking I might need to broadly monomorphise all parsing functions and lift error handling out of the hot call tree, whilst that might still be perf positive, it might not be necessary. This current monomorphized parseMessage is sufficient.